File: Server/src/main/java/pt/ist/ccu2021/queued/Server/dto/ScheduleDto.java
package pt.ist.ccu2021.queued.Server.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.*;
import pt.ist.ccu2021.queued.Server.domain.Day;
import pt.ist.ccu2021.queued.Server.domain.Schedule;

import java.sql.Time;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
@ToString
@JsonPropertyOrder({"day", "openingTime", "closingTime"})
public class ScheduleDto {

    @JsonProperty("day")
    private Day day;

    @JsonProperty("openingTime")
    private Time openingTime;

    @JsonProperty("closingTime")
    private Time closingTime;

    public ScheduleDto(Schedule schedule){
        day = Day.valueOf(schedule.getDay());
        openingTime = schedule.getOpeningTime();
        closingTime = schedule.getClosingTime();
    }

    public Schedule toDomain(int storeId){
        return Schedule.builder().day(day.toString()).storeId(storeId).openingTime(openingTime).closingTime(closingTime).build();
    }
}

File: Queued/lib/dto/ScheduleDto.dart
class ScheduleDto {
  ScheduleDto(this.day, this.openingTime, this.closingTime);

  String day;

  String openingTime;

  String closingTime;

  ScheduleDto.fromJson(Map<String, dynamic> json) {
    this.day = json['day'];
    this.openingTime = json['openingTime'];
    this.closingTime = json['closingTime'];
  }
}
